google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -366,9 +366,15 @@ public GapicSpannerRpc(final SpannerOptions options) {
       // If it is enabled in options uses the channel pool provided by the gRPC-GCP extension.
       maybeEnableGrpcGcpExtension(defaultChannelProviderBuilder, options);
 
+      InstantiatingGrpcChannelProvider defaultChannelProvider =
+          defaultChannelProviderBuilder.build();
```

Comment:
Can we make this conditional on whether `options.getChannelProvider()` returns null? That makes sure that we don't get any unwanted side-effects of creating a channel provider (that might not work) when the user has specified a custom channel provider.

Reply:
Added this check, please verify once.